exporter/awsemfexporter/datapoint.go

```diff
@@ -0,0 +1,296 @@
+	instrumentationLibraryName string
+	pdata.DoubleHistogramDataPointSlice
+}
+
+// DoubleSummaryDataPointSlice is a wrapper for pdata.DoubleSummaryDataPointSlice
+type DoubleSummaryDataPointSlice struct {
+	instrumentationLibraryName string
+	pdata.DoubleSummaryDataPointSlice
+}
+
+// At retrieves the IntDataPoint at the given index and performs rate calculation if necessary.
+func (dps IntDataPointSlice) At(i int) DataPoint {
+	metric := dps.IntDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap())
+	timestampMs := unixNanoToMilliseconds(metric.Timestamp())
+
+	var metricVal float64
+	metricVal = float64(metric.Value())
+	if dps.needsCalculateRate {
+		rateKey := createMetricKey(labels, dps.rateKeyParams)
+		rateTS := dps.timestampMs
+		if timestampMs > 0 {
+			// Use metric timestamp if available
+			rateTS = timestampMs
+		}
+		metricVal = calculateRate(rateKey, metricVal, rateTS)
+	}
+
+	return DataPoint{
+		Value:       metricVal,
+		Labels:      labels,
+		TimestampMs: timestampMs,
+	}
+}
+
+// At retrieves the DoubleDataPoint at the given index and performs rate calculation if necessary.
+func (dps DoubleDataPointSlice) At(i int) DataPoint {
+	metric := dps.DoubleDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap())
+	timestampMs := unixNanoToMilliseconds(metric.Timestamp())
+
+	var metricVal float64
+	metricVal = metric.Value()
+	if dps.needsCalculateRate {
+		rateKey := createMetricKey(labels, dps.rateKeyParams)
+		rateTS := dps.timestampMs
+		if timestampMs > 0 {
+			// Use metric timestamp if available
+			rateTS = timestampMs
+		}
+		metricVal = calculateRate(rateKey, metricVal, rateTS)
+	}
+
+	return DataPoint{
+		Value:       metricVal,
+		Labels:      labels,
+		TimestampMs: timestampMs,
+	}
+}
+
+// At retrieves the DoubleHistogramDataPoint at the given index.
+func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+	metric := dps.DoubleHistogramDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap())
+	timestamp := unixNanoToMilliseconds(metric.Timestamp())
+
+	return DataPoint{
+		Value: &CWMetricStats{
+			Count: metric.Count(),
+			Sum:   metric.Sum(),
+		},
+		Labels:      labels,
+		TimestampMs: timestamp,
+	}
+}
+
+// At retrieves the DoubleSummaryDataPoint at the given index.
+func (dps DoubleSummaryDataPointSlice) At(i int) DataPoint {
+	metric := dps.DoubleSummaryDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap())
+	timestampMs := unixNanoToMilliseconds(metric.Timestamp())
+
+	metricVal := &CWMetricStats{
+		Count: metric.Count(),
+		Sum:   metric.Sum(),
+	}
+	if quantileValues := metric.QuantileValues(); quantileValues.Len() > 0 {
+		metricVal.Min = quantileValues.At(0).Value()
+		metricVal.Max = quantileValues.At(quantileValues.Len() - 1).Value()
+	}
+
+	return DataPoint{
+		Value:       metricVal,
+		Labels:      labels,
+		TimestampMs: timestampMs,
+	}
+}
+
+// createLabels converts OTel StringMap labels to a map and optionally adds in the
```

Comment:
What is this "optional OTel instrumentation library name"?

Reply:
I should remove this line of comment. It was introduced in the closed PR(large one) which needs to use `instrumentation library` as key for grouping the metrics. `OTel instrumentation library dimension` is still handled in `metric_translator::buildCWMetric` function. I will add it back in next PR.